torch/csrc/jit/passes/peephole.cpp

```diff
@@ -360,7 +360,9 @@ struct PeepholeOptimizeImpl {
           node->output()->replaceAllUsesWith(output);
           changed_ = true;
         }
-      } else if (node->matches("aten::dim(Tensor self) -> int")) {
+        // disabled in onnx export to not constantify shape info
+      } else if (
+          node->matches("aten::dim(Tensor self) -> int") && !onnx_export_) {
```

Comment:
For this issue you should update the block for `aten::size(Tensor self) -> int[]`. Yet it appears that `aten::dim` would have the same issue, so this change is good as well.

Reply:
That's a separate but related issue, and I don't know if disabling this will break any models and cause this to be unable to land. I would prefer if it were handled in a different PR.